ndless-sdk/toolchain/build_toolchain.sh

```diff
@@ -30,6 +30,10 @@ OPTIONS_GCC="--target=$TARGET --prefix=$PREFIX --enable-interwork --enable-multi
 OPTIONS_NEWLIB="--target=$TARGET --prefix=$PREFIX --enable-interwork --enable-multilib --with-gnu-as --with-gnu-ld --disable-newlib-may-supply-syscalls --disable-newlib-supplied-syscalls --with-float=soft --disable-werror --disable-nls --enable-newlib-io-float"
 OPTIONS_GDB="--target=$TARGET --prefix=$PREFIX --enable-interwork --enable-multilib --disable-werror --with-python"
 
+if [ ! $(gcc -v 2>&1 | grep clang > /dev/null) ]; then
```

Comment:
You test the output of *nothing* here, that just can't work.

Just do
```if (gcc -v 2>&1 | grep clang > /dev/null); then```

Looks better as well!